docs/api-reference/deck.md

```diff
@@ -121,11 +121,14 @@ Transitions between two viewState objects can also be achieved by providing set
 
 If `initialViewState` is provided, the `Deck` component will track view state changes from any attached `controller` using internal state, with `initialViewState` as its initial view state.
 
+If the `initialViewState` prop changes, the internally tracked view state will be updated to match the new "initial" view state.
+
 Notes:
 
 * The `props.onViewStateChange` callback will still be called, if provided.
 * If `props.viewState` is supplied by the application, the supplied `viewState` will always be used, "shadowing" the `Deck` component's internal `viewState`.
-* In simple applications, use of the `initialViewState` prop can avoid the need track the view state in the application .
+* In simple applications, use of the `initialViewState` prop can avoid the need track the view state in the application.
+* Any new `initialViewState` value is compared shallowly with the previous value. Therefore, it is especially important for React applications to use a constant or component state as the `initialViewState` value, in order to avoid any unexpected "reset" of the camera.
```

Comment:
I think the shallow compare should be discussed. For the pydeck/json case, where the app builds new JSON payloads to control the visualization, the shallow equality will always be `false`.

I think the best compromise might be to compare `initialViewState` (if supplied) deeply.